### PR TITLE
Fix bolt handling

### PR DESCRIPTION
**Pull request by D-Cysteine**
This PR includes two fixes for part replacement:
 * When checking for different materials, bolt head material will now be checked. Previously, only shaft material was checked, so it was not possible to replace a bolt core when changing only the head material.
 * Fix for incorrect damage calculation in new tool for projectiles (arrows, bolts, javelins, etc.). Previously, the `"BaseAttack"` NBT tag was not getting copied over; this tag is used for projectile damage calculations. This should fix cases where replacing a bolt core resulted in a substantially lower-damage bolt than creating a new bolt.

Note that this PR will break part replacement (cause a crash to desktop) for any tools that are missing the `"BaseAttack"` NBT tag. That tag was added in this commit in 2015:
 * https://github.com/GTNewHorizons/TinkersConstruct/commit/283c3cf12a09aa1c6ca2ef2435a0519b863ccb41

so tools created before that commit will break. Let me know if you'd like me to add in a check for this case.

### CODE REVIEW

**Comment by Prometheus0000** [comment]
So this fixes replacing bolt parts? If so, could you make a reminder issue, or modify the related quest, because the quest tells people it's broken, and they're not going to know it's fixed. Also maybe an announcement in upcoming features.

**Comment by D-Cysteine** [comment]
Thanks! Done in:
 * https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/pull/8512

Should we hold off on announcing the fix until after this PR is approved / merged?

**Comment by Prometheus0000** [comment]
Probably? I doubt it'd take long though.

**Review by Dream-Master — APPROVED** [review]

**Pull request merged**
